core/app/models/spree/stock_location.rb

```diff
@@ -99,6 +101,16 @@ def fill_status(variant, quantity)
       end
     end
 
+    def self.default
+      Rails.cache.fetch("#{Rails.application.class.parent_name.underscore}_default_stock_location") do
+        default_location = where(default: true).first
+        if default_location.blank? && (default_location = first)
```

Comment:
`.blank?` is a too powerful primitive here. `.nil?` does for this case equivalent work but is part of core.